src/main/java/com/northconcepts/datapipeline/foundations/examples/jdbc/JoinCsvFiles.java

```diff
@@ -0,0 +1,87 @@
+package com.northconcepts.datapipeline.foundations.examples.jdbc;
+
+import com.northconcepts.datapipeline.core.DataReader;
+import com.northconcepts.datapipeline.core.DataWriter;
+import com.northconcepts.datapipeline.core.StreamWriter;
+import com.northconcepts.datapipeline.csv.CSVReader;
+import com.northconcepts.datapipeline.csv.CSVWriter;
+import com.northconcepts.datapipeline.jdbc.JdbcConnectionFactory;
+import com.northconcepts.datapipeline.jdbc.JdbcReader;
+import com.northconcepts.datapipeline.jdbc.JdbcWriter;
+import com.northconcepts.datapipeline.jdbc.sql.select.Select;
+import com.northconcepts.datapipeline.job.Job;
+import com.northconcepts.datapipeline.transform.BasicFieldTransformer;
+import com.northconcepts.datapipeline.transform.TransformingReader;
+
+import java.io.File;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+
+public class JoinCsvFiles {
+
+    public static void main(String[] args) throws Throwable {
+
+        Connection connection = JdbcConnectionFactory.wrap("org.h2.Driver", "jdbc:h2:mem:jdbcTableSort;MODE=MySQL", "sa", "").createConnection();
+        createCreditBalanceTable(connection);
+
+        DataWriter writer = new JdbcWriter(connection, "CreditBalance");
+        DataReader reader = new CSVReader(new File("example/data/input/credit-balance-insert-records2.csv")).setFieldNamesInFirstRow(true);
+        Job job1 = Job.runAsync(reader, writer);
+
+
+        connection = JdbcConnectionFactory.wrap("org.h2.Driver", "jdbc:h2:mem:jdbcTableSort;MODE=MySQL", "sa", "").createConnection();
+        createAccountTable(connection);
+
+        writer = new JdbcWriter(JdbcConnectionFactory.wrap("org.h2.Driver", "jdbc:h2:mem:jdbcTableSort;MODE=MySQL", "sa", ""), "Account");
+        reader = new CSVReader(new File("example/data/input/user_account.csv")).setFieldNamesInFirstRow(true);
+        reader = new TransformingReader(reader).add(new BasicFieldTransformer("DoB").stringToDate("yyyy-MM-dd"));
```

Comment:
Can we leave all fields as strings?

Reply:
Since we're trying to showcase a dynamic process, can we use `SchemaTransformer` instead of hardcoding the conversions?

Please see if something like https://northconcepts.com/docs/examples/transform-records-using-schema will work.